### PR TITLE
Fix micro step logging

### PR DESCRIPTION
**Pull request by mikasenghaas**
```bash
uv run sft @ configs/debug/sft.toml --log.level debug
```

**Previous**

We reported the scaled micro step loss

Screenshot 2025-09-13 at 9 22 59 PM

**Now**

We report the unscaled micro step loss, which matches the batch loss scale

Screenshot 2025-09-13 at 9 24 41 PM

---

**GitHub Issue**: #908 
**Linear Issue**: Resolves PRIMERL-99

### CODE REVIEW

**Review by Jackmin801 — APPROVED** [review]
makes sense. lgtm!

File: src/prime_rl/trainer/sft/train.py [hunk 1]
```diff
@@ -183,24 +183,22 @@ def train(config: SFTTrainerConfig):
...
+                # Compute average loss over unmasked tokens
                 loss = loss[loss_mask].mean()
 
-                # Scale loss by number of gradient accumulation steps
-                loss /= grad_accum_steps
-
-                # Accumulate loss
-                batch_loss += loss
+                # Accumulate average loss over gradient accumulation steps
+                batch_loss += loss / grad_accum_steps
```
Comment (samsja) [inline src/prime_rl/trainer/sft/train.py]:
```suggestion
                batch_loss += loss.detach() / grad_accum_steps
```
otherwise we will keep this in the graph but never backward on it

**Review by samsja — CHANGES_REQUESTED** [review]
hmm not we can do this more elegantly.

```python

loss = loss / grad_acc
loss.backward()

batch_loss += loss.detach()

**Comment by mikasenghaas** [comment]
> hmm not we can do this more elegantly.
> 
> ```python
> loss = loss / grad_acc
> loss.backward()
> 
> batch_loss += loss.detach()
> ```

yea i thought about doing this but then when reporting you have to do `loss.item() * grad_acc_steps` which i'd argue is even uglier. in the current setup we maintain the definition that `loss` (and `max_vio`) are the *micro-step metrics*, i.e. they have the same scale as the `batch_loss` and `batch_max_vio` at all times so we can use them for reporting as is. we only scale when we need to, e.g. when accumulating into batch metrics or before calling .backward() which we do inline. i acc prefer it this way but if you have really strong opinions we can change ofc

**Comment by samsja** [comment]
> > hmm not we can do this more elegantly.
> > ```python
> > loss = loss / grad_acc
> > loss.backward()
> > 
> > batch_loss += loss.detach()
> > ```
> 
> yea i thought about doing this but then when reporting you have to do `loss.item() * grad_acc_steps` which i'd argue is even uglier. in the current setup we maintain the definition that `loss` (and `max_vio`) are the _micro-step metrics_, i.e. they have the same scale as the `batch_loss` and `batch_max_vio` at all times so we can use them for reporting as is. we only scale when we need to, e.g. when accumulating into batch metrics or before calling .backward() which we do inline. i acc prefer it this way but if you have really strong opinions we can change ofc

okay fair, with the detach like you added then its good

**Review by samsja — APPROVED** [review]
lgtm
